Don't detect nwjs host as running in nodejs

timfish opened the pull request:
#53

timfish committed:
don't detect nwjs host as running in nodejs

niieani commented on src/index.js:
```diff
@@ -5,7 +5,7 @@ let bootstrapQueue = [];
 let Aurelia = null;
 const host = PLATFORM.global;
-const isNodeLike = typeof process !== 'undefined' && !process.browser;
+const isNodeLike = typeof process !== 'undefined' && !process.browser && !process.versions['nw'];
```
I think it would be better to do the change here instead:
https://github.com/timfish/bootstrapper/blob/6cb43ef2c87c0e776acb4360fbc540b9f4936650/src/index.js#L87

Rename `isElectronRenderer` to something like `isRenderer` and set it to true if we're either inside an Electron Renderer or an NwJS instance.

EisenbergEffect replied:
@timfish Are you able to update the PR based on that recommendation?

timfish committed:
@niieani had a better idea

Changed to look for `process.versions['node-webkit']`. I asked on gitter
and this is more backwards compatible

EisenbergEffect commented:
@niieani Can you do one final quick review? If it's good, I can merge and get a release out asap.

niieani commented on src/index.js:
```diff
@@ -84,9 +84,9 @@ function createLoader() {
   let type;
 
-  const isElectronRenderer = isNodeLike && process.type === 'renderer';
+  var isRenderer = isNodeLike && (process.type === 'renderer' || process.versions['node-webkit']);
```
If you could do one last change, and use `const` instead of `var` I'd really appreciate it :)
We try not to use `var` where possible.

Looks good otherwise!

niieani replied:
One more questions, is it: 
`versions['node-webkit']`
or
`versions['nw']`
?

Asking, cause you used both in the two of your commits.

timfish replied:
I use const too so no idea how I managed that. 

Both `versions['node-webkit']` and `versions['nw']` work for me. I asked on the nwjs gitter channel and `node-webkit` is more backwards compatible so works with versions < 0.12.3 too.

timfish committed:
silly mistake

niieani reviewed (approved):
Thanks, looks great now!

EisenbergEffect commented:
Fix is merged and released. 2.0.1
